internal/controller/sync.go

```diff
@@ -487,3 +498,34 @@ func certDurationFromRoute(r *routev1.Route) (time.Duration, error) {
 	}
 	return duration, nil
 }
+
+// This function returns the hostnames that have been admitted by an Ingress Controller
+// Usually this is just `.spec.host`, but as of OpenShift 4.11 users may also specify `.spec.subdomain`,
+// in which case the fully qualified hostname is derived from the hostname of the Ingress Controller.
+// In both cases, the final hostname is reflected in `.status.ingress[].host`.
+func getRouteHostnames(r *routev1.Route) []string {
```

Comment:
Do we really need to return a list of domains here? It seems like one Route = one host

Reply:
Yes we need, because a Route can be admitted by multiple ingress controllers. I had already prepared this example but forgot to attach it to this PR - my bad!

```yaml
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  creationTimestamp: "2023-05-03T07:10:21Z"
  name: test-subdomain
  namespace: openshift-cern-ingress-health-checks
  resourceVersion: "860465"
  uid: b6f0e564-ce65-43c8-8bb4-d989e6d76ed4
spec:
  subdomain: test-subdomain
  tls:
    termination: edge
  to:
    kind: Service
    name: health-checks-apps-shard-1
    weight: 100
  wildcardPolicy: None
status:
  ingress:
  - conditions:
    - lastTransitionTime: "2023-05-03T07:10:21Z"
      status: "True"
      type: Admitted
    host: test-subdomain.okd-apps-2.example.com
    routerCanonicalHostname: router-okd-apps-2.okd-apps-2.example.com
    routerName: okd-apps-2
    wildcardPolicy: None
  - conditions:
    - lastTransitionTime: "2023-05-03T07:10:21Z"
      status: "True"
      type: Admitted
    host: test-subdomain.okd-apps-1.example.com
    routerCanonicalHostname: router-okd-apps-1.okd-apps-1.example.com
    routerName: router-okd-apps-1
    wildcardPolicy: None
  - conditions:
    - lastTransitionTime: "2023-05-03T07:10:21Z"
      status: "True"
      type: Admitted
    host: test-subdomain.okd.example.com
    routerCanonicalHostname: router-default.okd.example.com
    routerName: default
    wildcardPolicy: None
```

This route is available at `test-subdomain.okd.example.com` as well as `test-subdomain.okd-apps-1.example.com`.

Reply:
Thanks, that totally makes sense.